model/histogram/float_histogram.go

```diff
@@ -263,9 +302,18 @@ func (h *FloatHistogram) Div(scalar float64) *FloatHistogram {
 //
 // The method reconciles differences in the zero threshold and in the schema, and
 // changes them if needed. The other histogram will not be modified in any case.
+// Adding is currently only supported between 2 exponential histograms, or between
+// 2 custom buckets histograms with the exact same custom bounds.
 //
 // This method returns a pointer to the receiving histogram for convenience.
-func (h *FloatHistogram) Add(other *FloatHistogram) *FloatHistogram {
+func (h *FloatHistogram) Add(other *FloatHistogram) (*FloatHistogram, error) {
+	if h.HasCustomBounds() != other.HasCustomBounds() {
+		return nil, ErrHistogramsIncompatibleSchema
+	}
+	if h.HasCustomBounds() && !floatBucketsMatch(h.CustomBounds, other.CustomBounds) {
```

Comment:
I'd move this check into the `if` on line 355, because that's where we depend on it and want to make sure the schemas align before calling addBuckets

Reply:
discussed IRL, better to fail fast